### DISCUSSION TITLE
Using Cloudflare to bypass Cloudflare (potential vulnerability; already public)

### DISCUSSION THREAD

**Discussion by Simonh98** [homeassistant-apps/app-cloudflared#491] [Q&A]
Note in advance: I intentionally did not follow the security guidelines, because the problem mentioned in this post is related to Cloudflare itself and is already public. If this is still not desired, please delete this post.

Hi all, 
I came across [this German article](https://certitude.consulting/blog/de/cloudflare-verwenden-um-cloudflare-zu-umgehen/) which identifies a potential Cloudflare security vulnerability due to Cloudflare's cross-user trust in their own certificates. This allows other Cloudflare users who also use Cloudflare's own certificate to establish legitimate and trusted connections to other users' servers. 

Unfortunately, I'm not good at assessing the risk myself, so how do you guys see the risk of using it with this Homeassistant addon and with Homeassistant in general? Cloudflare itself, according to this article, closed the issue by calling it "informative" and apparently hasn't done anything since.

**Comment by brenner-tobias** [comment]
Thanks a lot for raising this.
I did check the article and I assume that the mentioned issues are not relevant for our use-case.
The reason behind this is the fact, that we are only using Cloudflare Tunnel for connections to our HA instance. So any incoming connection via "normal" WAN is blocked anyways by the Firewall of the users and only the traffic from inside our very own tunnel is accepted. The tunnels themselves are not affected by this issue but rather other ways of verifying that traffic is allowed (e.g. by whitelisting Cloudflare Server IPs).
